app/forms/queries/projects/form.rb

```diff
@@ -52,8 +52,13 @@ class Queries::Projects::Create < ApplicationForm
         label: I18n.t(:button_cancel),
         tag: :a,
         data: { "params-from-query-target": "anchor" },
-        href: OpenProject::StaticRouting::StaticUrlHelpers.new.projects_path
+        href: projects_path(query_id: @query)
       )
     end
   end
+
+  def initialize(query:)
+    super()
+    @query = query
```

Comment:
I didn't find a way to get model (query) from the ApplicationForm, so simpler and probably cleaner to just pass it